src/std/io/bio/util.ss

```diff
@@ -653,3 +655,20 @@
 
 (def (expt-cache-get len)
   (vector-ref +expt-cache+ (fx- len 1)))
+
+(def (read-available-u8 reader start: (start 0) end: (end #f))
```

Comment:
Define these as reader extensions, so that we can dot them and have unchecked variants.

`(defreader-ext (read-avaliable-u8...))`.

Also drop the keyword args, it is inconsistent with ptetty much everything.